pkg/api/folder.go

```diff
@@ -52,7 +52,7 @@ func (hs *HTTPServer) GetFolderByID(c *models.ReqContext) response.Response {
 	if err != nil {
 		return response.Error(http.StatusBadRequest, "id is invalid", err)
 	}
-	folder, err := hs.folderService.GetFolderByID(c.Req.Context(), c.SignedInUser, c.OrgId, id)
+	folder, err := hs.folderService.GetFolderByID(c.Req.Context(), c.SignedInUser, id, c.OrgId)
```

Comment:
Should a test have failed? This is serious enough that it prevents users to upgrade to 8.5: https://github.com/grafana/terraform-provider-grafana/issues/463

Reply:
There are tests, but I think this would only have been caught by an end to end test, and it doesn't seem like we have any for these endpoints. I agree that it'd be nice to have tests that would've caught this, maybe something for the abstractions team to look into?

Reply:
🤔 How would that test have been constructed? AFAICT, this would work fine for the admin user since that would've called for (1, 1). Alternatively, having types for OrgID and UserID would've allowed the type system to pick this up during compile-time, and that would be true throughout Grafana, but that would be a fairly large project.

Reply:
Yup, that's what I would do as well - create two folders and query for the second one. But I imagine that we have a lot of other methods for querying by ID that also aren't tested in this way.

Out of curiosity, are we planning to deprecate querying by ID in favour of querying by UID?